node.go

```diff
@@ -156,6 +156,20 @@ func (opt *optionalNode) Repeated() bool       { return false }
 func (opt *optionalNode) Required() bool       { return false }
 func (opt *optionalNode) GoType() reflect.Type { return reflect.PtrTo(opt.Node.GoType()) }
 
+// FieldID wraps a node to provide node field id
+func FieldID(node Node, id int32) Node { return &fieldIDNode{Node: node, id: id} }
+
+type fieldIDNode struct {
+	Node
+	id int32
+}
+
+type nodeID interface {
+	ID() int32
+}
+
+func (f *fieldIDNode) ID() int32 { return f.id }
+
```

Comment:
For consistency, should we use int here as well?
```suggestion
// FieldID wraps a node to provide node field id
func FieldID(node Node, id int) Node { return &fieldIDNode{Node: node, id: id} }

type fieldIDNode struct {
	Node
	id int
}

type nodeID interface {
	ID() int
}

func (f *fieldIDNode) ID() int { return f.id }

```

Reply:
done a9c4cfd